pkg/steps/pod.go

```diff
@@ -185,16 +151,91 @@ func TestStep(config api.TestStepConfiguration, resources api.ResourceConfigurat
 		podClient,
 		artifactDir,
 		jobSpec,
+		testSecret,
 	)
 }
 
-func PodStep(name string, config PodStepConfiguration, resources api.ResourceConfiguration, podClient PodClient, artifactDir string, jobSpec *api.JobSpec) api.Step {
+func PodStep(name string, config PodStepConfiguration, resources api.ResourceConfiguration, podClient PodClient, artifactDir string, jobSpec *api.JobSpec, testSecret string) api.Step {
 	return &podStep{
 		name:        name,
 		config:      config,
 		resources:   resources,
 		podClient:   podClient,
 		artifactDir: artifactDir,
 		jobSpec:     jobSpec,
+		secret:      testSecret,
+	}
+}
+
+func (s *podStep) getPodObject() (*coreapi.Pod, error) {
+	image := fmt.Sprintf("%s:%s", s.config.From.Name, s.config.From.Tag)
+	containerResources, err := resourcesFor(s.resources.RequirementsForStep(s.config.As))
+	if err != nil {
+		return nil, fmt.Errorf("unable to calculate %s pod resources for %s: %s", s.name, s.config.As, err)
+	}
+
+	if len(s.config.From.Namespace) > 0 {
+		return nil, fmt.Errorf("pod step does not supported an image stream tag reference outside the namespace")
+	}
+
+	return &coreapi.Pod{
+		ObjectMeta: meta.ObjectMeta{
+			Name: s.config.As,
+			Labels: trimLabels(map[string]string{
+				PersistsLabel:    "false",
+				JobLabel:         s.jobSpec.Job,
+				BuildIdLabel:     s.jobSpec.BuildId,
+				ProwJobIdLabel:   s.jobSpec.ProwJobID,
+				CreatedByCILabel: "true",
+			}),
+			Annotations: map[string]string{
+				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
+				annotationContainersForSubTestResults: s.name,
+			},
+		},
+		Spec: coreapi.PodSpec{
+			ServiceAccountName: s.config.ServiceAccountName,
+			RestartPolicy:      coreapi.RestartPolicyNever,
+			Containers: []coreapi.Container{
+				{
+					Name:                     s.name,
+					Image:                    image,
+					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
+					Resources:                containerResources,
+					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
+					VolumeMounts:             podVolume(s.secret),
+				},
+			},
+			Volumes: podSecret(s.secret),
+		},
+	}, nil
+}
+
+func podSecret(name string) []coreapi.Volume {
+	if len(name) > 0 {
+		return []coreapi.Volume{
+			{
+				Name: "secret",
+				VolumeSource: coreapi.VolumeSource{
+					Secret: &coreapi.SecretVolumeSource{
+						SecretName: name,
+					},
+				},
+			},
+		}
+	}
+	return []coreapi.Volume{}
+}
+
+func podVolume(name string) []coreapi.VolumeMount {
+	if len(name) > 0 {
+		return []coreapi.VolumeMount{
+			{
+				Name:      "secret",
+				ReadOnly:  true,
+				MountPath: "/usr/secrets",
```

Comment:
I like delimiters idea better than a new flag. I don't expect this to be widely used either way.

Reply:
Personally, I don't mind about these hardcodings, they make usage simple, we do not need to expose every single element to be configurable.

I would like to see the `"secret"` name to be extracted to a constant (because it needs to match in `podSecret` and `podVolume` and it might not be clear to someone who reads this a month from now) and _maaaaybe_ call the mountpoint something less less generic (`ci-operator-secrets`? `configured-secrets`?) to reduce conflict possibility. But that's probably just bikeshedding.